test/replication-luatest/gh_10088_apply_deletion_of_replica_from_cluster_on_deleted_replica_test.lua

```diff
@@ -0,0 +1,129 @@
+local fio = require('fio')
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g_one_member_cluster = t.group('one_member_cluster')
+local g_three_member_cluster = t.group('three_member_cluster')
+
+g_one_member_cluster.before_all(function(cg)
+    cg.master = server:new{alias = 'master'}
+    cg.master:start()
+    cg.replica = server:new{alias = 'replica', box_cfg = {
+        replication = cg.master.net_box_uri,
+        replication_synchro_timeout = 120,
+    }}
+    cg.replica:start()
+    -- Add the replica to the master's replicaset table: needed for testing
+    -- the behaviour of the deleted replica after recovery.
+    cg.master:update_box_cfg{replication = cg.replica.net_box_uri}
+    cg.master:exec(function()
+        t.helpers.retrying({timeout = 120}, function()
+            t.assert_not_equals(box.info.status, "orphan")
+        end)
+    end)
+    -- Make `_cluster` space synchronous.
+    cg.master:exec(function()
+        box.ctl.promote()
+        box.space._cluster:alter{is_sync = true}
```

Comment:
Are you sure `box.ctl.promote()` after completion guarantees the instance is writable? I think it might be still read-only for short time after that. Is the test stable if you run it 50 times in parallel 16 workers, for example?

Reply:
The instances are read-write by default (I don't use elections), and AFAIC, the instance owns the limbo after `box.ctl.promote` returns, so there is a guarantee it can add entries to it, i.e., it is synchro-writable.

AFAIC, the test is stable.